Source/WebCore/html/parser/HTMLParserIdioms.cpp

```diff
@@ -107,7 +102,7 @@ double parseToDoubleForNumberType(StringView string, double fallbackValue)
         return fallbackValue;
 
     bool valid = false;
-    double value = string.toDouble(valid);
+    double value = string.toDouble(valid, true);
```

Comment:
Be careful of the side effect on HTMLInputElement in the Number state, as pointed out by https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#floating-point-numbers:valid-floating-point-number-3
IIUC, that one will use that parseToDoubleForNumberType function and expects it to reject invalid values, (including those with leading whitespace, leading plus sign, and/or trailing junk)
https://github.com/WebKit/WebKit/blob/29eeaffe3de46d4c072decbcacb10f99ffffb925/Source/WebCore/html/NumberInputType.cpp#L263-L268
But they should be caught by WPT html/semantics/forms/the-input-element/number.html if I'm reading correctly.

Reply:
Thanks — I changed that `parseToDoubleForNumberType(proposedValue)` call in `NumberInputType::sanitizeValue` to `parseValidHTMLFloatingPointNumber(proposedValue).value()`

But I can’t yet get test results for that `html/semantics/forms/the-input-element/number.html` WPT test, because when I run a build from this PR branch against it, the build crashes — and I’ve not yet been able to get a stack trace from the crash, so it’s going to take me some time to debug it. (The EWS results here show that it’s causing crashes with a couple dozen other test files as well.)